### DISCUSSION TITLE
ckb 0.111.0 (aaa5158 2023-09-14)

### DISCUSSION THREAD

**Discussion by doitian** [nervosnetwork/ckb#4157] [Announcements]
### Highlights

- This version will activate ckb2023 hardfork in the dev chains.
- There's an ongoing softfork activation since [v0.110.1](https://github.com/nervosnetwork/ckb/releases/tag/v0.110.1).

Major Changes

- Consensus changes:
    - CKB 2023 edition released (#3917)
    - `epoch_duration_target` now affects epoch length in Dummy mode (#4097)
- RPC breaking changes:
    - Set `jsonrpc::Ratio` as the type of threshold field for DeploymentsInfo and Deployment (#3980)
- CLI breaking change:
    - `ckb init` creates config files even when an unsupported spec is specified (#4012)

### Compatibility Table

| Network | Consensus Version        | Minimal CKB Version |
|---------|------------------------- | ------------------- |
| dev     | ckb2023 since epoch 0    |                     |
| testnet | ckb2021 since epoch 3113 | v0.101.0            |
| mainnet | ckb2021 since epoch 5414 | v0.103.0            |

See [how to connect to various chains](https://github.com/nervosnetwork/ckb/wiki/Chains) in Wiki.

### Downloads

| OS                | Arch  | Runtime Dependencies | Package                 | Sign                |
| ----------------- | ----- | -------------------- | ----------------------- | ------------------- |
| macOS             | x64   | macOS 10.15 or above | [zip][macos-zip]        | [PGP][macos-pgp]    |
| macOS  (Portable) | x64   | macOS 10.15 or above | [zip][macos-p-zip]      | [PGP][macos-p-pgp]  |
| macOS             | ARM64 | macOS 10.15 or above | [zip][macosa-zip]       | [PGP][macosa-pgp]   |
| macOS  (Portable) | ARM64 | macOS 10.15 or above | [zip][macosa-p-zip]     | [PGP][macosa-p-pgp] |
| Linux             | x64   | glibc, libstdc++     | [tarball][linux-zip]    | [PGP][linux-pgp]    |
| Linux  (Portable) | x64   | glibc, libstdc++     | [tarball][linux-p-zip]  | [PGP][linux-p-pgp]  |
| CentOS            | x64   | glibc, libstdc++     | [tarball][centos-zip]   | [PGP][centos-pgp]   |
| CentOS (Portable) | x64   | glibc, libstdc++     | [tarball][centos-p-zip] | [PGP][centos-p-pgp] |
| Windows           | x64   | VC++ Redistributable | [zip][windows-zip]      | [PGP][windows-pgp]  |
| Linux             | ARM64 | glibc, libstdc++     | [tarball][arm-zip]      | [PGP][arm-pgp]      |
| Docker            |       |                      | [dockerhub][]           |                     |

The `.asc` files are signatures. It is wise and more secure to check out for the files [integrity](https://github.com/nervosnetwork/ckb/blob/develop/docs/integrity-check.md).

Try the portable variants if the binaries complains about illegal instructions.

[macos-zip]: https://github.com/nervosnetwork/ckb/releases/download/v0.111.0/ckb_v0.111.0_x86_64-apple-darwin.zip
[macos-pgp]: https://github.com/nervosnetwork/ckb/releases/download/v0.111.0/ckb_v0.111.0_x86_64-apple-darwin.zip.asc
[macos-p-zip]: https://github.com/nervosnetwork/ckb/releases/download/v0.111.0/ckb_v0.111.0_x86_64-apple-darwin-portable.zip
[macos-p-pgp]: https://github.com/nervosnetwork/ckb/releases/download/v0.111.0/ckb_v0.111.0_x86_64-apple-darwin-portable.zip.asc
[macosa-zip]: https://github.com/nervosnetwork/ckb/releases/download/v0.111.0/ckb_v0.111.0_aarch64-apple-darwin.zip
[macosa-pgp]: https://github.com/nervosnetwork/ckb/releases/download/v0.111.0/ckb_v0.111.0_aarch64-apple-darwin.zip.asc
[macosa-p-zip]: https://github.com/nervosnetwork/ckb/releases/download/v0.111.0/ckb_v0.111.0_aarch64-apple-darwin-portable.zip
[macosa-p-pgp]: https://github.com/nervosnetwork/ckb/releases/download/v0.111.0/ckb_v0.111.0_aarch64-apple-darwin-portable.zip.asc
[linux-zip]: https://github.com/nervosnetwork/ckb/releases/download/v0.111.0/ckb_v0.111.0_x86_64-unknown-linux-gnu.tar.gz
[linux-pgp]: https://github.com/nervosnetwork/ckb/releases/download/v0.111.0/ckb_v0.111.0_x86_64-unknown-linux-gnu.tar.gz.asc
[linux-p-zip]: https://github.com/nervosnetwork/ckb/releases/download/v0.111.0/ckb_v0.111.0_x86_64-unknown-linux-gnu-portable.tar.gz
[linux-p-pgp]: https://github.com/nervosnetwork/ckb/releases/download/v0.111.0/ckb_v0.111.0_x86_64-unknown-linux-gnu-portable.tar.gz.asc
[centos-zip]: https://github.com/nervosnetwork/ckb/releases/download/v0.111.0/ckb_v0.111.0_x86_64-unknown-centos-gnu.tar.gz
[centos-pgp]: https://github.com/nervosnetwork/ckb/releases/download/v0.111.0/ckb_v0.111.0_x86_64-unknown-centos-gnu.tar.gz.asc
[centos-p-zip]: https://github.com/nervosnetwork/ckb/releases/download/v0.111.0/ckb_v0.111.0_x86_64-unknown-centos-gnu-portable.tar.gz
[centos-p-pgp]: https://github.com/nervosnetwork/ckb/releases/download/v0.111.0/ckb_v0.111.0_x86_64-unknown-centos-gnu-portable.tar.gz.asc
[windows-zip]: https://github.com/nervosnetwork/ckb/releases/download/v0.111.0/ckb_v0.111.0_x86_64-pc-windows-msvc.zip
[windows-pgp]: https://github.com/nervosnetwork/ckb/releases/download/v0.111.0/ckb_v0.111.0_x86_64-pc-windows-msvc.zip.asc
[arm-zip]: https://github.com/nervosnetwork/ckb/releases/download/v0.111.0/ckb_v0.111.0_aarch64-unknown-linux-gnu.tar.gz
[arm-pgp]: https://github.com/nervosnetwork/ckb/releases/download/v0.111.0/ckb_v0.111.0_aarch64-unknown-linux-gnu.tar.gz.asc
[dockerhub]: https://hub.docker.com/r/nervos/ckb/tags

#### Features

-   #3917 **consensus: (BREAKING)** CKB 2023 edition (@zhangsoledad)
-   #3963 **rpc(Chain):** Add `only_committed` param for `get_transaction` (@eval-exec)
-   #3980 **rpc: (BREAKING)** Set `jsonrpc::Ratio` as the type of `threshold` field for `DeploymentsInfo` and `Deployment` (@eval-exec)
-   #3988: Update ckb-vm to v0.24.0 (@mohanson)
-   #3996: Suspend support in spawn syscalls (@xxuejie)
-   #4002: Let crate `ckb-hash` support `no-std` (@yangby-cryptape)
-   #4097 **consensus: (BREAKING)** `epoch_duration_target` now affects epoch length in `Dummy` mode (@eval-exec)
-   #4126: Notify dummy miner for new work (@quake)

#### Improvements

-   #3970 **perf:** Replace sync struct HeaderView with HeaderIndexView (@quake)
-   #4000: Reduce `PeerStore::load_from_dir_or_default` timecost (@eval-exec)
-   #4003: Derive implements Clone for cell (@liya2017)
-   #4073: Split ckb-gen-types from ckb-types (@EthanYuan)
-   #4123 **vm(spawn):** Calucate the correct cycles_base (@mohanson)
-   #4127 **vm(spawn):** Always consume extra_cycles (@mohanson)
-   #4129: Fix parse `RelayV3`'s message name for metrics service (@eval-exec)
-   #4133: Keep peer store address unique (@driftluo)

#### Bug Fixes

-   #4011 **vm:** `set_debug_printer` should updates generator's `debug_printer` (@mohanson)
-   #4012 **cli: (BREAKING)** `ckb init` creates config files even when an unsupported spec is specified. (@eval-exec)
-   #4015 Fix permanent difficulty mode reward (@zhangsoledad)
-   #4030 Fix RUSTSEC-2023-0044 warning (@eval-exec)
-   #4151: Add `derive(Hash)` to `ScriptHashType` (@eval-exec)

<hr /><em>This discussion was created from the release <a href='https://github.com/nervosnetwork/ckb/releases/tag/v0.111.0'>ckb 0.111.0 (aaa5158 2023-09-14)</a>.</em>

**Comment by joshyates1980** [comment]
@quake In reference to: [this issue](https://github.com/nervosnetwork/ckb/discussions/4147#discussioncomment-7007791), I deleted the entire "ckb 0.111.0-rc12" folder because of local computer issues.

I downloaded "ckb 0.111.0" and executed: **ckb init** and then **ckb run**.

I have a new "node_id":"QmVgA1wDAcFJQ12WaS9LQ7wbgrQUfzjqAQ772b5ACetBtu" as of yesterday afternoon so hopefully after some time the CKB Full Node will be fully operational and display on [CKB Node Probe](https://nodes.ckb.dev/)

You mentioned that `ckb-indexer` has been migrated into ckb. Does this mean I do not need to execute `ckb run --indexer` when I execute Neuron wallet? I'm going to wait until the full node is downloaded before I try Neuron again on this particular computer.

**Reply by doitian** [reply]
@Keith-CY can you help to answer the question of Neuron?

**Reply by Keith-CY** [reply]
Hi @joshyates1980 
`ckb-indexer` has been integrated into `ckb`, so you don't have to run them both, but the `indexer` module of `ckb` is not enabled by default, so you still need to run `ckb` by `ckb run --indexer` if it's expected to be an external node of Neuron.

Neuron lauches the built-in node with `--indexer` option, too, ref: https://github.com/nervosnetwork/neuron/blob/9d14ac6e1fe11a80c719c4adb7499cd8af7888a1/packages/neuron-wallet/src/services/ckb-runner.ts#L96

**Comment by joshyates1980** [comment]
@Keith-CY and @quake : It seems **_Neuron-v0.110.2_** will need to be updated before `ckb run --indexer` will sync with **_ckb_v0.111.0_**. Is this correct?

![20230920](https://github.com/nervosnetwork/ckb/assets/7255545/0ed0593f-f9e8-4102-ae01-8dcaf8dad64e)

**Reply by Keith-CY** [reply]
The compatibility validation is pretty simple that it just checks if the version number of an external CKB node is the same as that of the builtin one, so connecting to a CKB node released after a specific neuron will be alerted because the version numbers are definitely different.

We have introduced a compatibility table in the next version for precise validation. But this case(connecting to an external node released after Neuron) cannot be avoided because the table will be packaged into Neuron instead of being fetched at runtime.

BTW, I have learned that CKB@v0.111.0 has incompatible updates but we haven’t tested it with Neuron, so CKB@v0.110.2 will be set as the builtin node.


Ref:
- Compatibility table: https://github.com/nervosnetwork/neuron/blob/develop/compatible.csv
- Builtin node of Neuron@v0.110.3: https://github.com/nervosnetwork/neuron/pull/2845

**Comment by doitian** [comment]
Light Client Protocol Softfork Activation in Mainnet

| Field                | Value | Note                    |
| -------------------- | ----- | ----------------------- |
| start                | 8,282 | 2023/09/01 00:00:00 utc |
| timeout              | 8,552 | 8,282 + 270             |
| min_activation_epoch | 8,648 | 2023/11/01 00:00:00 utc |
| threshold            | 80%   |                         |

20 hours remaining to reach the min activation epoch.
